cmd/modelcmd/modelcommand.go

```diff
@@ -381,12 +419,51 @@ func (w *modelCommandWrapper) inner() cmd.Command {
 	return w.ModelCommand
 }
 
+// IAASOnlyCommand is used as a marker and is embedded
+// by commands which should only run in IAAS models.
+type IAASOnlyCommand interface {
+	_iaasonly() // not implemented, marker only.
+}
+
+// validateIAASOnyCommand returns an error if an IAAS-only command
+// is run on a CAAS model.
+func (w *modelCommandWrapper) validateIAASOnyCommand(runStarted bool) error {
```

Comment:
Typo - Ony instead of Only

Reply:
Actually, I think this should be called validateModelType - a caller shouldn't care what this method checks specifically (that it's an IAAS model if the command is IAAS-only), just that it is checking that the model type is right for the command.

Reply:
How about "validateCommandForModelType"